### PR TITLE
Correctly check db dir existence (fixes #1753)

### LINKED ISSUE #1753: Crash on upgrade to v1.19.0 on old androids due to recursive deletion utility
There's a crash related to the recursive deletion utility function used, occurring on old (I think <=7) androids only:

```
java.lang.NoSuchMethodError: 
  at org.apache.commons.io.FileUtils.forceDelete (FileUtils.java:1341)
  at org.apache.commons.io.FileUtils.cleanDirectory (FileUtils.java:324)
  at org.apache.commons.io.FileUtils.deleteQuietly (FileUtils.java:1236)
  at com.nutomic.syncthingandroid.activities.FirstStartActivity.upgradeToApiLevel30 (FirstStartActivity.java:213)
  at com.nutomic.syncthingandroid.activities.FirstStartActivity.access$500 (FirstStartActivity.java:51)
  at com.nutomic.syncthingandroid.activities.FirstStartActivity$ViewPagerAdapter$3.onClick (FirstStartActivity.java:333)
  at android.view.View.performClick (View.java:6312)
  at android.widget.TextView.performClick (TextView.java:11202)
  at android.view.View$PerformClick.run (View.java:23985)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6816)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1563)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1451)
```

We should do the deletion manually instead.

### PR DESCRIPTION
**Pull request by imsodin**

### CODE REVIEW

File: app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java [hunk 1]
```diff
@@ -218,7 +218,7 @@ private void upgradeToApiLevel30() {
             } catch (Throwable e) {
                 Log.w(TAG, "Deleting database with FileUtils failed", e);
                 Util.runShellCommand("rm -r" + dbDir.getAbsolutePath(), false);
-                if (dbDir.exists()) {
```
Comment (AudriusButkevicius) [inline app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java:221]: The -r above looks like its missing a space

File: app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java [hunk 2]
```diff
@@ -218,7 +218,7 @@ private void upgradeToApiLevel30() {
...
+                if (new File(this.getFilesDir(), "index-v0.14.0.db").exists()) {
```
[the remark this answers is not in the document]
Comment (imsodin) [reply app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java:221]: I was too lazy to check if the information is created at construction time or when calling `.exists()`, thus created a new object to be certain.

[the remark this answers is not in the document]
Comment (imsodin) [reply app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java]: Removed it again, it's pretty clear from docs that `File` is just a "hollow" representation of something at this path, not doing any filesystem operations on creation or any other caching of state.

**Comment by imsodin** [comment]
@AudriusButkevicius Could you check again if this looks good from your point of view now?

**Comment by imsodin** [comment]
I am merging to go for another round of hotfix testing as this is currently affecting users, so can't really become worse :)

**Pull request merged**